packages/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -119,33 +122,79 @@ export class ConnectionPropsForSessCfg {
+
+            if (ConnectionPropsForSessCfg.propHasValue(finalSessCfg.password)=== false) {
+                promptForValues.push("password");
+            }
+
+            // put all the needed properties in an array and call the external function
+            const answer = optsToUse.getAllValuesBack(promptForValues);
+
+            // validate what values are given back and move it to finalSessCfg
+            if (ConnectionPropsForSessCfg.propHasValue(answer.hostname)) {
+                finalSessCfg.hostname = answer.hostname;
+            }
+            if (ConnectionPropsForSessCfg.propHasValue(answer.port)) {
+                finalSessCfg.port = answer.port;
+            }
+            if (ConnectionPropsForSessCfg.propHasValue(answer.user)) {
+                finalSessCfg.user = answer.user;
+            }
+            if (ConnectionPropsForSessCfg.propHasValue(answer.password)) {
+                finalSessCfg.password = answer.password;
+            }
+        }
+
         // if our caller permits, prompt for host and port as needed
         if (optsToUse.doPrompting) {
             if (ConnectionPropsForSessCfg.propHasValue(finalSessCfg.hostname) === false) {
                 let answer = "";
-                while (answer === "") {
-                    answer = await CliUtils.promptWithTimeout(
-                        "Enter the host name of your service: "
-                    );
-                    if (answer === null) {
-                        throw new ImperativeError({msg: "Timed out waiting for host name."});
+                if (optsToUse.getValuesBack) {
+                    answer = optsToUse.getValuesBack("hostname");
+                } else {
+                    while (answer === "") {
+                        answer = await CliUtils.promptWithTimeout(
+                            "Enter the host name of your service: "
+                        );
+                        if (answer === null) {
+                            throw new ImperativeError({msg: "Timed out waiting for host name."});
+                        }
                     }
                 }
                 finalSessCfg.hostname = answer;
             }
 
             if (ConnectionPropsForSessCfg.propHasValue(finalSessCfg.port) === false) {
                 let answer: any;
-                while (answer === undefined) {
-                    answer = await CliUtils.promptWithTimeout(
-                        "Enter the port number for your service: "
-                    );
-                    if (answer === null) {
-                        throw new ImperativeError({msg: "Timed out waiting for port number."});
-                    } else {
-                        answer = Number(answer);
-                        if (isNaN(answer)) {
-                            throw new ImperativeError({msg: "Specified port was not a number."});
+                if (optsToUse.getValuesBack) {
+                    answer = optsToUse.getValuesBack("port");
```

Comment:
Should getValueBack/getAllValuesBack be an async function that returns a promise and can be awaited like `CliUtils.promptWithTimeout`?

Reply:
yeah. why not. i'll add that